Modules/Loadable/Colors/MRMLDM/vtkMRMLColorBarDisplayableManager.cxx

```diff
@@ -0,0 +1,386 @@
+
+  Program: 3D Slicer
+
+  Copyright (c) Kitware Inc.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+==============================================================================*/
+
+// MRMLDisplayableManager includes
+#include "vtkMRMLColorBarDisplayableManager.h"
+
+// MRML includes
+#include <vtkMRMLApplicationLogic.h>
+#include <vtkMRMLColorNode.h>
+#include <vtkMRMLDisplayNode.h>
+#include <vtkMRMLInteractionEventData.h>
+#include <vtkMRMLInteractionNode.h>
+#include <vtkMRMLLightBoxRendererManagerProxy.h>
+#include <vtkMRMLScene.h>
+#include <vtkMRMLSliceLogic.h>
+#include <vtkMRMLSliceNode.h>
+#include <vtkMRMLViewNode.h>
+#include <vtkMRMLScalarVolumeNode.h>
+#include <vtkMRMLScalarVolumeDisplayNode.h>
+#include <vtkMRMLColorNode.h>
+#include <vtkMRMLColorTableNode.h>
+#include <vtkMRMLProceduralColorNode.h>
+#include <vtkMRMLColorBarDisplayNode.h>
+
+#include <vtkSlicerScalarBarActor.h>
+
+// VTK includes
+#include <vtkActor2D.h>
+#include <vtkCallbackCommand.h>
+#include <vtkCellArray.h>
+#include <vtkMatrix4x4.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkPickingManager.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataMapper2D.h>
+#include <vtkProp.h>
+#include <vtkProperty2D.h>
+#include <vtkRenderer.h>
+#include <vtkRenderWindow.h>
+#include <vtkRenderWindowInteractor.h>
+#include <vtkSmartPointer.h>
+#include <vtkWeakPointer.h>
+#include <vtkVersion.h>
+#include <vtkScalarBarRepresentation.h>
+#include <vtkScalarBarWidget.h>
+#include <vtkLookupTable.h>
+
+// STD includes
+#include <iostream>
+#include <algorithm>
+#include <cstring>
+
+//---------------------------------------------------------------------------
+vtkStandardNewMacro(vtkMRMLColorBarDisplayableManager);
+
+//---------------------------------------------------------------------------
+class vtkMRMLColorBarDisplayableManager::vtkInternal
+{
+public:
+  vtkInternal(vtkMRMLColorBarDisplayableManager * external);
+  virtual ~vtkInternal();
+
+  vtkObserverManager* GetMRMLNodesObserverManager();
+  void Modified();
+
+  // Build/update color bar
+  void BuildColorBar();
+  void UpdateColorBar();
+
+  // Build/update actor and widget representation
+  void BuildActor();
+  void UpdateActor();
+
+  vtkSlicerScalarBarActor* GetActor() const { return this->ColorBarActor; }
+  vtkScalarBarWidget* GetWidget() const { return this->ColorBarWidget; }
+
+  vtkMRMLColorBarDisplayableManager* External;
+
+  vtkSmartPointer<vtkSlicerScalarBarActor> ColorBarActor;
+  vtkSmartPointer<vtkScalarBarWidget> ColorBarWidget;
```

Comment:
We should observe all vtkMRMLColorBarDisplayNodes here and add a vtkScalarBarWidget/vtkSlicerScalarBarActor for each that are displayable in the view. See vtkMRMLVolumeRenderingDisplayableManager and markups displayable manager for example.

Reply:
Got it! I will add support for multiple scalar bars, one for each displayable node.